obsproject/obs-studio
# UI: Create Properties & Log Viewer window XML files, migrate code

### Description

window-basic-properties was written in the earliest days of OBS. To make it easier to navigate & adjust (especially if we decide to make it more complex in the future), this moves all the structural code into a new .ui file and out of the C++ code.

This also modifies the ButtonBox in Filters to use matching buttons.

Additionally, this does the same for the Log Viewer window.

### Motivation and Context

Functional code goes in C++, structural code goes in XML.

My goal was mostly to make the constructors for these windows **much easier** to read. With 45 and 48 deletions respectively, I feel I have done so.

Additionally, anyone who wants to experiment with the UI using either Qt Designer or Qt Creator should be able to do so, with any OBS window.

I only realised after completing my changes that back in https://github.com/obsproject/obs-studio/commit/66f7330a586317501aed9ab7f1dc46732af6d57e the opposite was done, but I feel we understand Qt a lot more now and additionally the Properties dialog is complex enough that it is now worth doing.

### How Has This Been Tested?

* Launch the Properties and Log Viewer windows
* Notice they look the same as before

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
